docs/guide/ecosystem/editors-and-tools.md

```diff
@@ -0,0 +1,164 @@
+
+Vetur is being developed closely with VSCode and Vue core teams, to ensure maximum parity.
+
+<useful-links>
+<useful-links-section title="Useful Links">
+
+- [Official Website](https://code.visualstudio.com/)
+- [Vetur Plugin](https://marketplace.visualstudio.com/items?itemName=octref.vetur) - enable improved Vue support
+- [Vue VSCode Extension Pack](https://marketplace.visualstudio.com/items?itemName=sdras.vue-vscode-extensionpack) - collection of useful plugins for Vue app development
+
+</useful-links-section>
+</useful-links>
+
+### WebStorm/PhpStorm (JetBrains IDEs)
+
+If you need a fully featured IDE with all the bells and whistles, not just a code editor, WebStorm is a serious contender. Not free, JetBrains products like WebStorm and PhpStorm are feature rich IDEs, offering everything a web developer may need for their daily work. They have a free trial, free student licenses, as well as a free open beta testing program, which allows users to test JetBrains products early on.
+
+To enable Vue support you will need to install the official, free Vue.js plugin from the plugins menu. It will give you `.vue` file support with highlighting, `template` and `script` tag code completion, linting, component hinting, prop completion, type hinting, refactoring support and much more.
+
+As a fully featured IDE, WebStorm provides greater refactoring capabilities, improved code completion, hinting and highlighting, but at a cost of speed. It is heavy and can get choppy if you open too many files or projects at the same time.
+
+### CodeSandbox <badge text="Rising Star"/>
+
+CodeSandbox is the only online, cloud based code editor, that supports Vue's SFC format, among many other frameworks, popular client apps and server setups. Vue projects are built with Vue CLI, giving you a well known folder structure to extend upon. 
+
+Based on VSCode, CodeSandbox has a similar look and feel. When you create a Vue project, you will benefit from code completion, hinting, syntax highlighting and more. It even allows you to connect a sandbox to a Github repo and commit to it on each change. Apart from Vue CLI, they have templates for [Vuepress](../ecosystem/documentation.md#vuepress) and [Nuxt](../ecosystem/nuxt.md), just to name a few.
+
+CodeSandbox has a very generous freemium policy with 50 active sandboxes (projects) absolutely free.
+
+<useful-links>
+<useful-links-section title="Useful Links">
+
+- [Official Website](https://codesandbox.net)
+
+</useful-links-section>
+</useful-links>
+
+### Atom
+
```

Comment:
An amazing free, open-source code editor created by GitHub with plenty of extensions and superb user experience. With the available plugins, it supports a wide range of languages and features, including ones dedicated to Vue: smart code completion/suggestions, snippets, syntax highlighting, linting and formatting with both eslint and prettier, that works in template, script and style parts of the Vue SFC.

On top of that, Atom has great integration with GitHub when it comes to managing commits, PRs, reviewing changes and can be further extended with additional plugins. Some say its the most fashionable and good looking editor out there and the code completion is at the same time very smart (thanks to Tern.js), but also completely unintrusive. You’re the developer, you got this!

Some say, that it’s a bit slow, but that’s just a way of hinting that you should refactor your huge components.

<useful-links>
<useful-links-section title="Useful Links">

- [ternjs](https://github.com/tststs/atom-ternjs)
- [language-vue](https://github.com/hedefalk/atom-vue)
- [ide-vue](https://github.com/rwatts3/atom-ide-vue)
- [vue-fmt](https://github.com/voodeng/vue-fmt)

</useful-links-section>
</useful-links>

Reply:
Imho this section feels pretty opinionated. There are parts of the guide where we focus on opinions (like tips) but the ecosystem one should give an objective overview feel.

Reply:
It's written by the only one using it, what do you expect? 😆

Reply:
Oh. So why don’t we sort the elements in the guide alphabetically? Why does VS Code have the "hottest" icon? Why isn’t Atom listed in the "best" code editors? 
My point is, the whole thing is already opinionated. 😉 
The difference here is that I decided to actually add some personality into the description in hopes it makes for a more engaging content. If you don’t agree or like it, feel free to redact it – I won’t mind!